sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -81,6 +80,18 @@ public Response Response
         /// </summary>
         public TimeSpan? NetworkTimeout { get; set; }
 
+        internal void AddPolicies(RequestContext context)
+        {
+            if (context == null || context.Policies == null || context.Policies.Count == 0)
+            {
+                return;
+            }
+
+            Policies ??= new(context.Policies);
```

Comment:
Do we intentionally only set Policies on the first call to `AddPolicies`? If so, should we indicate somehow to the caller that it was ignored?

Reply:
It's an internal policy that should only be called once when the message is created.

Reply:
If it's important to only set once on creation, would it make more sense to implement this as an internal constructor and mark Policies readonly? If not, I think we should just set `Policies` here unconditionally.